### DISCUSSION TITLE
[DOC]Do I need the timestamp? Use of a Timeseriesgenerator, Keras

### DISCUSSION THREAD

**Discussion by Gradwanderer** [alan-turing-institute/sktime#820] [Q&A] [answered]
#### Describe the issue linked to the documentation
https://www.sktime.org/en/v0.4.2/examples/loading_data.html
 can't find a solution for my specific problem. Described here: https://www.sktime.org/en/v0.4.2/examples/loading_data.html
Do I need the timestamp? 

#### Suggest a potential alternative/fix

would be great to have an integration to timeseriesgenerators
This could be an easy comparable way for "classic" timeseries methods and neural nets.

Tell us how we could improve the documentation in this regard.
--> Perhaps there is a quick fix or workaround, that I havent found.

**Comment by mloning** [comment]
Thanks for reaching out @Gradwanderer - I'm not entirely sure what problem you're trying to solve or what your question is. 

There's an issue on data generators here: #353

**Comment by Gradwanderer** [comment, marked as answer]
May I just copie my slack Question here: 
I have a dataframe with some (35 columns) Variables and millions (rows) of timesteps. I would like to cut the data with the timeseriesgenerator (https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/sequence/TimeseriesGenerator). I have done it before for some neural nets. I would like to stick to the generator, because I need to skip some samples, but how many isn't clear at the moment. 

Is it possible to use the generator also to provide the examples for sktime?
I have a classification problem. The data is provided in a long mode. The generator does not provide timestamps. Or is there any workaround?
I would like to use sktime, but failed to get the data in shape. The data contains a time variable, but there was a problem while recording. So there are big gaps, but this was just a saving issue.

The Loaded date (reduced to 10 rows):

| var 1 | var2 | var3 | class 1 | class 2| class 3|
| ------- | -------- | ------ | ----------- |------ |------ |
|0.000565 | 0.97515|  0.000052 | 1 | 0 | 0|
|0.000558 | 0.97514 | 0.000044 |1 | 0 | 0|
|0.000553 | 0.97513 | 0.000036 |1 | 0 | 0|
|0.000547 | 0.97512 | 0.000028 |1 | 0 | 0|
|0.000541 | 0.97511 | 0.000018 |1 | 0 | 0|
|0.000536 | 0.97510 | 0.000010 |1 | 0 | 0|
|0.000531 | 0.97509 | 0.000006 |1 | 0 | 0|
|0.000525 | 0.97508 | 0.000013 |1 | 0 | 0|
|0.000520 | 0.97507 | 0.000021 |1 | 0 | 0|
|0.000515 | 0.97506 | 0.000030 |1 | 0 | 0|

Shape of the generator with samplelen of 6 and samplingrate of 1. It is the first sample. 

    Shape of the generator data and label: (1, 3, 3) (1, 3)
    [[[5.6456e-04 9.7515e-01 5.2059e-05]
      [5.5257e-04 9.7513e-01 3.6473e-05]
      [5.4131e-04 9.7511e-01 1.8311e-05]]]
    [[1 0 0]]

The second sample looks like:

    Shape of the generator data1 and label1: (1, 3, 3) (1, 3)
    [[[5.5834e-04 9.7514e-01 4.4129e-05]
      [5.4748e-04 9.7512e-01 2.8435e-05]
      [5.3617e-04 9.7510e-01 1.0407e-05]]]
    [[1 0 0]]

I need to split the time series, because the model should only get a time window of x values and should predict if there is an error state or not. I think the sktime package would be a great choice for "classic" methods. _

**Reply by mloning** [reply]
If in the above data, rows are time points, it looks like a time series annotation problem to me, which you can solve via time series classification by using a sliding window transformation. I would be helpful to provide a minimal, reproducible example of the problem that you're trying to solve.
